core/types/receipt.go

```diff
@@ -98,6 +101,7 @@ type storedReceiptRLP struct {
 	PostStateOrStatus []byte
 	CumulativeGasUsed uint64
 	Logs              []*LogForStorage
+	L1Fee             *big.Int
```

Comment:
is this the right place to add this field? what is the difference between the different receipt marshaling types?

Reply:
We could consider adding a new version: rename the old `storedReceiptRLP` to `v5StoredReceiptRLP`, add your new version as `storedReceiptRLP`, and update `DecodeRLP` accordingly. This way your change might be backward compatible.